src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -0,0 +1,454 @@
+        this.table = tableMeta.name;
+        this.partitionKeyType = tableMeta.partitionKeyType;
+        this.clusteringComparator = tableMeta.comparator;
+        this.target = TargetParser.parse(tableMeta, config);
+        this.config = config;
+        this.indexMetrics = new IndexMetrics(this, tableMeta);
+        this.validator = TypeUtil.cellValueType(target);
+
+        this.analyzerFactory = AbstractAnalyzer.fromOptions(getValidator(), config.options);
+
+        this.queryAnalyzerFactory = AbstractAnalyzer.hasQueryAnalyzer(config.options)
+                                    ? AbstractAnalyzer.fromOptionsQueryAnalyzer(getValidator(), config.options)
+                                    : this.analyzerFactory;
+        this.primaryKeyFactory = PrimaryKey.factory(tableMeta.comparator);
+    }
+
+    @VisibleForTesting
+    public IndexContext(String keyspace,
+                        String table,
+                        AbstractType<?> partitionKeyType,
+                        ClusteringComparator clusteringComparator,
+                        ColumnMetadata column,
+                        IndexMetadata config)
+    {
+        this.keyspace = keyspace;
+        this.table = table;
+        this.partitionKeyType = partitionKeyType;
+        this.clusteringComparator = clusteringComparator;
+        this.target = Pair.create(column, IndexTarget.Type.SIMPLE);
+        this.validator = column.type;
+        this.config = config;
+        this.indexMetrics = null;
+        Map<String, String> options = config != null ? config.options : Collections.emptyMap();
+        this.analyzerFactory = AbstractAnalyzer.fromOptions(getValidator(), options);
+        this.queryAnalyzerFactory = AbstractAnalyzer.hasQueryAnalyzer(options)
+                                    ? AbstractAnalyzer.fromOptionsQueryAnalyzer(getValidator(), options)
+                                    : this.analyzerFactory;
+        this.primaryKeyFactory = PrimaryKey.factory(clusteringComparator);
+    }
+
+    public IndexContext(TableMetadata table, ColumnMetadata column)
+    {
+        this.keyspace = table.keyspace;
+        this.table = table.name;
+        this.partitionKeyType = table.partitionKeyType;
+        this.clusteringComparator = table.comparator;
+        this.target = TargetParser.parse(table, column.name.toString());
+        this.validator = target == null ? null : TypeUtil.cellValueType(target);
+        this.indexMetrics = null;
+        this.config = null;
+        Map<String, String> options = Collections.emptyMap();
+        this.analyzerFactory = AbstractAnalyzer.fromOptions(getValidator(), options);
+        this.queryAnalyzerFactory = AbstractAnalyzer.hasQueryAnalyzer(options)
+                                    ? AbstractAnalyzer.fromOptionsQueryAnalyzer(getValidator(), options)
+                                    : this.analyzerFactory;
+        this.primaryKeyFactory = PrimaryKey.factory(clusteringComparator);
```

Comment:
I think this can be simplified to:
```
        this(table.keyspace,
             table.name,
             table.partitionKeyType,
             table.comparator,
             column,
             null);
```
That would reduce a bit the duplication between the three contructors.

Reply:
@adelapena We have recently introduced a change that has a single constructor for `IndexContext`. I would prefer that change because it reduces duplication and makes `null` checking of parameters easier.

I will make that change but please let me know if you don't think it's appropriate.

Reply:
Sure thing, any reduction on the duplication among constructors sounds good to me. A single constructor would be ideal.